core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/expiration/ThrottledUseAndTimeoutExpirationPolicy.java

```diff
@@ -53,15 +56,13 @@ public ThrottledUseAndTimeoutExpirationPolicy(@JsonProperty("timeToLive") final
     public boolean isExpired(final TicketState ticketState) {
         LOGGER.trace("Checking validity of ticket [{}]", ticketState);
         val lastTimeUsed = ticketState.getLastTimeUsed();
-        val currentTime = ZonedDateTime.now(ZoneOffset.UTC);
+        val currentTime = ZonedDateTime.now(clock);
```

Comment:
What's the difference between what exists and your change?

Reply:
In unit test, a fixed clock can be set to check the expiration of TGT at 0.99s(should be valid) or 1s(should be expired) after last use, this can not be handled by `Thread.sleep()` method.

Reply:
Like this in Spring source code:
https://github.com/spring-projects/spring-framework/blob/9277b470404ce1dce790fa82e684c9d8220940e7/spring-web/src/main/java/org/springframework/web/server/session/InMemoryWebSessionStore.java#L80-L93

Reply:
Thanks for the reference.